### PR TITLE
support for orchestrator running with multi instances

### PR DESCRIPTION
**Pull request by hoozecn**
#### What type of this PR

Add one of the following kinds:
/kind polish

#### What this PR does / why we need it:

The orchestrator need to run in singleton mode due to the cron jobs. 

Now we support it to run in "multiple instances" mode with the help of etcd features.

#### Which issue(s) this PR fixes:

- [orchestrator 单点问题修复](https://erda-org.erda.cloud/erda/dop/projects/387/issues/all?id=207975&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIiwiUkVTT0xWRUQiXSwiYXNzaWduZWVJRHMiOlsiMTAwMTM1NiJdfQ%3D%3D&issueTable__urlQuery=eyJwYWdlTm8iOjEsICJwYWdlU2l6ZSI6MTB9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6InRhYmxlIiwiY2hpbGRyZW5WYWx1ZSI6eyJrYW5iYW4iOiJkZWFkbGluZSJ9fQ%3D%3D&iterationID=506&type=TASK)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  Support for orchestrator running with multiple instances            |
| 🇨🇳 中文    |       支持  orchestrator 以多实例运行     |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/orchestrator/initialize.go [hunk 1]
```diff
@@ -65,15 +86,16 @@ func Initialize() error {
...
-		return err
-	}
-
-	logrus.Infof("start the service and listen on address: \"%s\"", conf.ListenAddr())
-	logrus.Errorf("[alert] starting orchestrator instance")
+	// Limit only one instance of scheduler to do the cron jobs
+	p.Election.OnLeader(func(ctx context.Context) {
+		logrus.Infof("i'm the leader now")
+		_ = initCron(ep, ctx)
+		logrus.Infof("i resign the leader now")
```
Comment (sfwn) [inline modules/orchestrator/initialize.go:95]: Do you need invoke Election.ResignLeader here?

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by sfwn** [comment]
add labels pls.

File: modules/orchestrator/initialize.go (hunk 1 above)
Comment (hoozecn) [reply modules/orchestrator/initialize.go:95]: Nope. The resign login has been done by the p.Election internally

Comment (hoozecn) [reply modules/orchestrator/initialize.go:95]: The ctx is closed after the resign.

**Comment by hoozecn** [comment]
> add labels pls.

done

**Review by QvodSoldier — APPROVED** [review]

**Comment by QvodSoldier** [comment]
Maybe we should add a TODO comment to tell others that we haven't really completed the multi-instance task. (Not very important, just a suggestion)

**Comment by hoozecn** [comment]
> Maybe we should add a TODO comment to tell others that we haven't really completed the multi-instance task. (Not very important, just a suggestion)

The cron jobs are required to be run in singleton mode. You may improve the deployment handler in distribution mode later if the performance is a REAL problem.

**Review by sfwn — APPROVED** [review]

**Pull request merged**